src/Sentry/Internal/Json.cs

```diff
@@ -1,14 +1,52 @@
+using System;
+using System.Collections;
 using System.IO;
+using System.Linq;
+using System.Reflection;
 using System.Text;
 using System.Threading;
 using System.Threading.Tasks;
 using Newtonsoft.Json;
 using Newtonsoft.Json.Converters;
+using Newtonsoft.Json.Serialization;
 
 namespace Sentry.Internal
 {
+    [AttributeUsage(AttributeTargets.Property)]
+    internal class DontSerializeEmptyAttribute : Attribute {}
+
     internal static class Json
     {
+        private class ContractResolver : DefaultContractResolver
+        {
+            protected override JsonProperty CreateProperty(MemberInfo member, MemberSerialization memberSerialization)
+            {
+                var jsonProperty = base.CreateProperty(member, memberSerialization);
+                var property = jsonProperty.DeclaringType.GetProperty(jsonProperty.UnderlyingName);
+
+                // DontSerializeEmpty
+                if (jsonProperty.ShouldSerialize is null &&
+                    property?.GetCustomAttribute<DontSerializeEmptyAttribute>() is {})
+                {
+                    // Collections
+                    if (property.PropertyType.GetInterfaces().Any(i => i == typeof(IEnumerable)))
```

Comment:
The previous code only serialized members annotated which were only the private, nullable fields, so it isn't true that the serializer called the properties accessors.

That said this is a lot more expensive than the previous code though it doesn't matter given it's anyway executed in the background thread and it's still negligent perf impact.

Reply:
True, I forgot it ignored non-annotated properties.